### ISSUE TITLE
To be more compatible with stdlib URI, needs a #to_s that is string URI that was parsed

### ISSUE THREAD

**Issue by jrochkind** [tlewin/ruby-uriparser#2]
Right now #to_s just gives you something opaque like `"#<UriParser::URI:0x00007f93443ab9c8>"`

Much existing code assumes it can turn a stdlib URI back into a string with #to_s.

**Comment by jrochkind** [comment]
Looks like it's intending to do this already....

https://github.com/tlewin/ruby-uriparser/blob/master/ext/uriparser_ext/uriparser.c#L392

I am not too familiar with C or C/ruby bridge, not sure how to debug.

**Comment by tlewin** [comment]
It was expected to return the original URI: https://github.com/tlewin/ruby-uriparser/blob/master/test/uriparser_test.rb#L86

Something might have changed in the C API. I'll check it.

**Comment by tlewin** [comment]
I've just checked the following snippet and it worked as expected:

```ruby
require 'uriparser'
require 'uriparser/uri_gem'

uri = URI.parse("https://github.com/tlewin/ruby-uriparser/issues/2")
puts uri
# https://github.com/tlewin/ruby-uriparser/issues/2
```

Could you please share the code here?

I am closing the issue, but if you're able to reproduce the error, then we can re-open.

**Comment by jrochkind** [comment]
Hi, thanks for responding. 

The difference reported is with calling the `#to_s` method on the thing returned from the parse, your example doesn't do that... although I can't reproduce what you post either!

I actually can't currently get `URI.parse` to be patched at all, not sure why, but I'll show you what I get with `URI()`.  

With the original:

```ruby
rirb(main):001:0> require 'uri'
=> false
irb(main):003:0> u = URI("http://example.com")
=> #<URI::HTTP http://example.com>
irb(main):004:0> u.to_s
=> "http://example.com"
```

With the uri_parser patch:

```ruby
irb(main):001:0> require 'uriparser'
=> true
irb(main):002:0> require 'uriparser/uri_gem'
=> true
irb(main):003:0>  u = URI("http://example.com")
=> #<UriParser::URI:0x00007fb73c894bf8>
irb(main):004:0> u.to_s
=> "#<UriParser::URI:0x00007fb73c894bf8>"
```

You can see that the thing returned from the parse, originally calling `to_s` on it returns the original URL as a string. With the parsed object from URIParser, calling `to_s` on it returns a sort of default ruby representation of an object, `"#<UriParser::URI:0x00007fb73c894bf8>"`

But if this gem is not entirely finished code from a long time ago and isn't currently maintained... fair enough! 

You of course aren't obligated to fix it, but you could perhaps leave this Issue open so someone else finding this gem and exploring would see that there is an issue with this.

**Comment by tlewin** [comment]
Which Ruby version are you using?

**Comment by jrochkind** [comment]
2.6.6.

**Comment by tlewin** [comment]
I figured out what's going on. The version with `to_s` implementation (cdd51803), was never published somehow. Sorry, my bad!

Anyway, yesterday I published a new version 0.2.0 and this should work fine.

**Comment by jrochkind** [comment]
OK, thanks!
